### PR TITLE
Add `pubspec.lock` for library package to `.gitignore`

### LINKED ISSUE #143: We should commit the `pubspec.lock` for library packages.
~See: https://dart.dev/guides/libraries/private-files#pubspeclock~

~Our library packages are located in `lib`.~

Edit: Renaming this from "We should not commit..." to "We should commit". Not committing a mono repo where a library package is _only_ used in the repository doesn't really make sense to not commit the `pubpsec.lock`. Not committing the `pubspec.lock` would result into some problems like every developer which have a different dependency tree and maybe is the app not working for this person.

### PR DESCRIPTION
**Pull request by nilsreichardt**
We should not include the `pubspec.lock` file for library packages. Source: https://dart.dev/guides/libraries/private-files#pubspeclock

Closes #143

### CODE REVIEW

**Comment by nilsreichardt** [comment]
> remote_configuration ⛔ Failure
Exception: Could not run dart pub global run tuneup check --fail-on-todos (exit code 1): 
>
> stdout:Checking project remote_configuration...
>
>  info - 'package:firebase/firebase.dart' is deprecated and shouldn't be used. This package will receive no further updates.> > at lib/src/implementation/firebase_web_remote_configuration.dart:9:1 - (deprecated_member_use)
>
> 1 issue found; analyzed 7 source files in 17.1s.

This requires to upgrade the al Firebase packages to latest state.

**Comment by nilsreichardt** [comment]
This PR is blocked by some other tasks (e.g. https://github.com/SharezoneApp/sharezone-app/pull/218#issuecomment-1123924298).

File: lib/.gitignore [hunk 1]
```diff
@@ -5,4 +5,8 @@
 
 # Files created by Flutter
 **/.flutter-plugins
-**/.flutter-plugins-dependencies
\ No newline at end of file
+**/.flutter-plugins-dependencies
+
+# We should not include the pubspec.lock file for library packages. Source:
+# https://dart.dev/guides/libraries/private-files#pubspeclock
+pubspec.lock
```
[the remark this answers is not in the document]
Comment (nilsreichardt) [reply lib/.gitignore:12]: Both solutions work

[the remark this answers is not in the document]
Comment (Jonas-Sander) [reply lib/.gitignore:12]: Ah okay, thanks:)  
Then lets keep it how it is.

**Review by Jonas-Sander — APPROVED** [review]

**Comment by nilsreichardt** [comment]
I'm going to close this PR. I don't I will continue to work on this in the near future.

**Pull request closed without merging**